app/src/main/java/com/example/wnews/views/home/news/NewsProvider.kt

```diff
@@ -0,0 +1,156 @@
+package com.example.wnews.views.home.news
+
+import com.example.wnews.models.LikeResponse
+import com.example.wnews.models.ListNewsResponse
+import com.example.wnews.models.News
+import com.example.wnews.models.UserAuth
+import com.example.wnews.providers.RetrofitProvider
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+object NewsProvider {
+
+
+    var arrayListNews = arrayListOf<News>()
+
+    var isLoading = false
+
+    lateinit var userAuthProvider: UserAuth
+
+    lateinit var newsViewProvider: NewsView
+
+    fun initNewsProvider(userAuth: UserAuth, newsView: NewsView) {
+        userAuthProvider = userAuth
+        newsViewProvider = newsView
+    }
+
+    fun onResponseUpdateNewsLike(newsId: Int, newsView: NewsView) {
+
+        val call: Call<LikeResponse> = RetrofitProvider.newsService.putLike(
+            newsId,
+            userAuthProvider.token,
+            userAuthProvider.client,
+            userAuthProvider.uid
+        )
+        call.enqueue(object : Callback<LikeResponse?> {
+
+            override fun onResponse(
+                call: Call<LikeResponse?>?,
+                response: Response<LikeResponse?>?
+            ) {
+
+                if (response!!.isSuccessful) {
+
+                    newsView.onSuccessResponse()
+
+                } else if (response.errorBody() != null) {
+                    newsView.onFailureResponse(response.message())
+                }
+
+
+            }
+
+            override fun onFailure(call: Call<LikeResponse?>?, t: Throwable?) {
+                newsView.onFailureResponse("")
```

Comment:
As a recommendation for the future (is not necessary to change this in this PR), API calls callbacks may be separate into 3 different callbacks to avoid passing empty values like this in the callback.

`newsView.onResponseSuccess` -> API request successfully made, server response OK (response code `2xx`)
`newsView.onResponseFailure` -> API request successfully made, server response error (response code != `2xx`)
`newsView.onRequestFailure` -> API request failed on origin, server wasn't even reached.